core/payment-driver/dummy/src/lib.rs

```diff
@@ -54,65 +57,40 @@ impl PaymentDriver for DummyDriver {
     fn get_account_balance<'a>(
         &'a self,
         _address: &str,
-    ) -> Pin<Box<dyn Future<Output = Result<AccountBalance, PaymentDriverError>> + 'static>> {
+    ) -> Pin<Box<dyn Future<Output = Result<BigDecimal, PaymentDriverError>> + 'static>> {
         let amount = "1000000000000000000000000";
-        Box::pin(future::ready(Ok(AccountBalance {
-            base_currency: Balance {
-                currency: Currency::Gnt,
-                amount: utils::str_to_big_dec(&amount).unwrap(),
-            },
-            gas: Some(Balance {
-                currency: Currency::Eth,
-                amount: utils::str_to_big_dec(&amount).unwrap(),
-            }),
-        })))
+        Box::pin(future::ready(Ok(utils::str_to_big_dec(&amount).unwrap())))
     }
 
     fn schedule_payment<'a>(
         &self,
-        invoice_id: &str,
-        amount: PaymentAmount,
+        amount: BigDecimal,
         sender: &str,
         recipient: &str,
         _due_date: DateTime<Utc>,
-    ) -> Pin<Box<dyn Future<Output = Result<(), PaymentDriverError>> + 'a>> {
+    ) -> Pin<Box<dyn Future<Output = Result<String, PaymentDriverError>> + 'a>> {
         let payments = self.payments.clone();
         let details = PaymentDetails {
             recipient: recipient.to_string(),
             sender: sender.to_string(),
-            amount: amount.base_currency_amount,
+            amount: amount,
             date: Some(Utc::now()),
         };
-        let invoice_id = invoice_id.to_string();
+        // FIXME: generate payment_order_id?
```

Comment:
2 questions on payment_order_id:
- Where should it be generated? just a new UUID?
- Should it be part of `PaymentDetails`?

Reply:
Just a new UUID. Even a counter incremented with every new order should do.

Nope, it doesn't need to go to payment details.

Reply:
updated it to be an incremental counter